tests/FsCheck.Test/Arbitrary.fs

```diff
@@ -260,9 +260,11 @@ module Arbitrary =
     [<Property>]
     let ``DateTime shrinks seconds, minutes and hours`` (value:DateTime) =
         shrink value
-        |> Seq.forall (fun v -> v.Second = 0 
-                                 || (v.Second = 0 && v.Minute = 0) 
-                                 || (v.Second = 0 && v.Minute = 0 && v.Hour = 0) )
+        |> Seq.forall (fun v -> v.Kind = DateTimeKind.Unspecified   // TODO: all the cases below are redundant?
```

Comment:
Indeed they are. A stronger check would be to check that if the hours are 0, then also minutes, seconds & kind are 0 resp. unspecified, and so on.

Reply:
Not sure that would work, since it's possible that the original value happened to have `Hour` equal to zero.  I don't think there is much that can be asserted here except `v.Kind = DateTimeKind.Unspecified`, given the implementation of the shrinker.